src/home/components/ChangePasswordContents/CurrentPasswordForm/CurrentPasswordForm.tsx

```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react';
+
+import { BoxButton } from '@yourssu/design-system-react';
+import { useNavigate } from 'react-router-dom';
+
+import { getPassword } from '@/home/apis/getPassword';
+import { PasswordInput } from '@/home/components/ChangePasswordContents/PasswordInput/PasswordInput';
+import { sessionTokenType } from '@/home/types/GetPassword.type';
+import { api } from '@/service/TokenService';
+
+import {
+  StyledInputContainer,
+  StyledBoxContainer,
+  StyledTitle,
+  StyledInputTitle,
+  StyledButtonContainer,
+} from './CurrentPasswordForm.style';
+
+interface CurrentPasswordFormProps {
+  onConfirm: () => void;
+  setSessionToken: ({ sessionToken }: sessionTokenType) => void;
+}
+
+export const CurrentPasswordForm = (Props: CurrentPasswordFormProps) => {
+  const { onConfirm, setSessionToken } = Props;
+  const [currentPassword, setCurrentPassword] = useState('');
+  const [isError, setIsError] = useState(false);
+  const navigate = useNavigate();
+
+  const checkCurrentPassword = (currentPassword: string) => {
+    const accessToken = api.getAccessToken();
+    if (!accessToken) {
+      alert('로그인이 필요합니다.');
+      navigate('/');
+    }
+
+    getPassword({ password: currentPassword, accessToken }).then((res) => {
+      if (!res) navigate('/Login');
+      if (res?.match) {
+        setIsError(false);
+        setSessionToken(res.sessionToken as sessionTokenType);
+        onConfirm();
+      } else {
+        setIsError(true);
+      }
+    });
+  };
```

Comment:
```suggestion
  const checkCurrentPassword = async (currentPassword: string) => {
    const accessToken = api.getAccessToken();
    if (!accessToken) {
      alert('로그인이 필요합니다.');
      navigate('/');
      return;
    }

    const passwordMatchData = await getPassword({ password: currentPassword, accessToken });
    
    if (!passwordMatchData) {
      navigate('/Login');
      return;
    }
    
    if (!passwordMatchData.match) {
      setIsError(true);
      return;
    }
    
    setIsError(false);
    setSessionToken(passwordMatchData.sessionToken as sessionTokenType);
    onConfirm();
  };
```

이런식으로 바꾸시는 건 어떠신가요?

1. `navigate()` 호출 이후, `return` 을 넣어주어 직관적으로 함수를 탈출하도록 이해를 돕습니다
2. `Promise` -> `async/await` 로 변경하였습니다.
3. `else` 를 제거했습니다.

Reply:
추가로 더 논의하면 좋을 점이 있는데,

`home/apis/getPassword.ts` 의 `getPassword()` 의 역할이 무엇인지 애매하다고 느낍니다.

- 저 함수명만 놓고 보았을 때, 무엇의 password를 가져오는지 애매합니다. (애초에 password를 가져오는 것도 아님)
- 무엇보다도 코드 상 password를 가져오는지, password가 같은지 매칭하는 건지 함수를 까기 전까지는 애매함.

=> 따라서 `getPassword` 의 파일명과 함수명을 바꿔야된다고 생각합니다.

- `getUserPasswordMatch`
- `matchUserPassword`

등이 있겠네요

Reply:
저는 getUserPasswordMatch 에 한 표 던질래요

Reply:
getUserPasswordMatch로 선정되었습니다.
9c730732a4b523295a44cd3148476b97fd672e8a